assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -633,21 +635,31 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
+
+      // Helper for checking webhook objects in the buildConfig spec and creating if missing.
+      function reuseOrCreateWebhook(type) {
+        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === type})
+        if (_.isEmpty(webhooks)) {
+          webhooks = {
```

Comment:
@jwforres no just an array of objects.
Check https://paste.fedoraproject.org/358316/14612661/